app/controllers/listings/applications_controller.rb

```diff
@@ -13,6 +13,9 @@ def index
     end
 
     def new
+      if @listing.Lottery_Status != 'Not Yet Run'
+        redirect_to "/listings/#{@listing.Id}"
```

Comment:
You could use the semantic `listing_url` + param as opposed to the interpolation. Also might be cleaner to make this a before action.

Reply:
Updated